src/compass/utils/validate_product.py

```diff
@@ -0,0 +1,333 @@
+'''
+Collection of functions to compare 2 CSLC HDF5 contents and metadata
+'''
+import argparse
+import os
+
+import h5py
+import numpy as np
+from osgeo import gdal
+
+
+DATA_ROOT = 'science/SENTINEL1'
+
+
+def cmd_line_parser():
+    """
+    Command line parser
+    """
+    parser = argparse.ArgumentParser(
+        description="""Validate reference and generated (secondary) S1 CSLC products""",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument(
+        '-r', '--ref-product', type=str, dest='ref_product',
+        help='Reference CSLC or static layer product (i.e., golden dataset)')
+    parser.add_argument(
+        '-s', '--sec-product', type=str, dest='sec_product',
+        help='Secondary CSLC or static layer product to compare with reference')
+    parser.add_argument('-p', '--product-type', type=str, dest='product_type',
+                        choices=['CSLC', 'static_layer'],
+                        default='CSLC', help='Type of file to be validated')
+    return parser.parse_args()
+
+
+def _grid_nfo_retrieve(path_h5, dataset_names, is_static_layer):
```

Comment:
Question: What does `nfo` mean in the function name?

Reply:
`nfo` was me being lazy and not using `info`. I will correct for clarity.